iOS_SDK/OneSignalSDK/Source/OSMessagingController.m

```diff
@@ -280,14 +284,64 @@ - (void)showAndImpressMessage:(OSInAppMessage *)message {
     });
 }
 
+- (void)messageViewPageImpressionRequest:(OSInAppMessage *)message withPageId:(NSString *)pageId {
+    if (message.isPreview) {
+        [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:[NSString stringWithFormat:@"Not sending page impression for preview message. ID: %@",pageId]];
+        return;
+    }
+    
+    if (!pageId) {
+        [OneSignal onesignal_Log:ONE_S_LL_ERROR message:[NSString stringWithFormat:@"Attempting to send page impression for nil page id"]];
+        return;
+    }
+    
+    NSString *messagePrefixedPageId = [message.messageId stringByAppendingString:pageId];
+        
+    if ([[message getViewedPageIds] containsObject:pageId] || [self.viewedPageIDs containsObject:messagePrefixedPageId]) {
```

Comment:
sry is not clear to me why we have pageIds under the message and a global list

Reply:
I removed it from the message!